modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -102,31 +110,20 @@ func (k Keeper) OnChanCloseConfirm(
 	return nil
 }
 
-// validateControllerPortParams asserts the provided connection sequence and counterparty connection sequence
-// match that of the associated connection stored in state
-func (k Keeper) validateControllerPortParams(ctx sdk.Context, connectionHops []string, connectionSeq, counterpartyConnectionSeq uint64) error {
+// validateConnectionParams asserts the provided controller and host connection identifiers match that of the associated connection stored in state
+func (k Keeper) validateConnectionParams(ctx sdk.Context, connectionHops []string, controllerConnectionID, hostConnectionID string) error {
```

Comment:
I am thinking of moving this function to pkg `types` which would look something like:

```
func ValidateConnectionParams(ctx sdk.Context, channelKeeper ChannelKeeper, connectionHops []string, connectionID, counterpartyConnectionID string) error {
        connID := connectionHops[0]
	connection, err := channelKeeper.GetConnection(ctx, connectionID)
	if err != nil {
		return err
	}

	if connectionID != connID {
		return sdkerrors.Wrapf(connectiontypes.ErrInvalidConnection, "expected %s, got %s", connID, connectionID)
	}

	if counterpartyConnectionID != connection.GetCounterparty().GetConnectionID() {
		return sdkerrors.Wrapf(connectiontypes.ErrInvalidConnection, "expected %s, got %s", connection.GetCounterparty().GetConnectionID(), counterpartyConnectionID)
	}

}
```

This would be then be reusable in both `controller` and `host` submodules, passing the appropriate controller/host connection IDs in place of `connectionID` and `counterpartyConnectionID`.
If all are aligned on this, I'm happy to do it in a follow up PR.